modules/google-analytics/classes/class-jetpack-google-amp-analytics.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * Add support for Google Analytics e-commerce events for AMP pages.
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Bail if accessed directly
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Jetpack_Google_AMP_Analytics class.
+ */
+class Jetpack_Google_AMP_Analytics {
+	/**
+	 * Add hooks.
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_add_to_cart', array( $this, 'amp_add_to_cart' ), 10, 6 );
+		add_action( 'woocommerce_thankyou', array( $this, 'amp_after_purchase' ), 10, 1 );
+		add_action( 'wp_footer', array( $this, 'amp_send_ga_events' ) );
+	}
+
+	/**
+	 * Generate a GA event when adding an item to the cart.
+	 *
+	 * @param string $cart_item_key Cart item key.
+	 * @param string $product_id Product ID.
+	 * @param int    $quantity Product quantity.
+	 * @param int    $variation_id Product variation ID.
+	 * @param object $variation Product variation.
+	 * @param object $cart_item_data Cart item data.
+	 */
+	public function amp_add_to_cart( $cart_item_key, $product_id, $quantity, $variation_id, $variation, $cart_item_data ) {
+		if ( ! class_exists( 'Jetpack_AMP_Support' ) || ! Jetpack_AMP_Support::is_amp_request() ) {
+			return;
+		}
+
```

Comment:
Should we check for `Jetpack_Google_Analytics_Options::track_add_to_cart_is_enabled()` before we do anything here?